File: linked_queue.go
package main

import "fmt"

type Node struct {
	value int
	next  *Node
}

type LinkedQueue struct {
	head *Node
	tail *Node
	size int
}

func (lq *LinkedQueue) enqueue(val int) {
	newest := &Node{value: val}
	if lq.size == 0 {
		lq.head = newest

	} else {
		lq.tail.next = newest
	}
	lq.tail = newest
	lq.size++
}

func (lq *LinkedQueue) dequeue() {
	if lq.size == 0 {
		fmt.Println("queue is empty..")
	} else {
		fmt.Println("first out is :", lq.head.value)
		lq.head = lq.head.next
		lq.size--
	}

}
func (lq *LinkedQueue) display() {
	ptr := lq.head
	for i := 0; i < lq.size; i++ {
		fmt.Printf("%v-->", ptr.value)
		ptr = ptr.next
	}

}

func main() {
	lq := LinkedQueue{}
	lq.enqueue(1)
	lq.enqueue(2)
	lq.enqueue(3)
	lq.enqueue(4)
	lq.dequeue()
	lq.display()

}

File: stacks/linked_queue.py
class Node:
    __slots__="_element","_next"
    def __init__(self,element,next):
        self._element=element
        self._next=next

class LinkedQueue:
    def __init__(self):
        self.head=None
        self.tail=None
        self.size=0
    
    def __len__(self):
        return (self.size)
    
    def isEmpty(self):
        return self.size==0

    def enqueue(self,e):
        newest=Node(e,None)
        if self.isEmpty():
            self.head=newest
        else:
            self.tail._next=newest
        self.tail=newest 
        self.size+=1
    
    def dequeue(self):
        if self.isEmpty():
            print("queue is empty")
            return 
        else:
            # print("first out is :",self.head._element)
            e=self.head._element
            self.head=self.head._next
        self.size-=1
        return e
    def display(self):
        p=self.head
        while p:
            print(p._element,end="-->")
            p=p._next
        print()


# lq=LinkedQueue()
# lq.enqueue(1)
# lq.enqueue(2)
# lq.enqueue(3)
# lq.enqueue(4)
# lq.dequeue()
# lq.display()
